src/ResourceManagement/AppService/HostNameSslBindingImpl.cs

```diff
@@ -66,6 +66,18 @@ private string GetCertificateThumbprint(string pfxPath, string password)
             throw new KeyNotFoundException("Private key not found in the pfx file");
         }
 
+        private static string GetCertificateThumbprint(byte[] pfxByteArray, string password)
+        {
+            var collection = new X509Certificate2Collection();
```

Comment:
Why not just use `new X509Certificate2(rawData, password)` to get the thumbprint, rather than loading it into a collection first?

Reply:
I've followed the existing implementation in `GetCertificateThumbprint(string pfxPath, string password)` as I was unsure if there was some nuance to the particular approach previously taken.

Happy to update my method and the existing one if there is consensus that the `X509Certificate2Collection` is unnecessary.

Reply:
Ah OK. I was just curious, as I came across this PR while trying to see if there was a existing issue for a problem I'm having related to certificates, and the constructor approach is what I do in my particular app.

Reply:
On closer inspection I don't think the existing implementation is actually behaving as intended - enumerating a X509Certificate2Collection import doesn't provide any guarantees about the presence of a private key

Thanks for highlighting this martin, I'll put something together to address.